$OpenBSD$
- respect ports CFLAGS
- use OpenBSD shared library naming
- upstream endianness fix since v1.14 in master branch.
- replace modp_b64 base64 en/decode implementation with ISC one from OpenBSD
--- Makefile.orig	Thu Sep 12 00:28:58 2013
+++ Makefile	Thu Jan  9 12:20:31 2014
@@ -5,23 +5,25 @@ MAKE_DIR     = install -d
 INSTALL_DATA = install
 
 CC=gcc
-CFLAGS=-O2 -Wall -g -D_FORTIFY_SOURCE=2 -fstack-protector -fPIC
+CFLAGS=-Wall -g -D_FORTIFY_SOURCE=2 -fstack-protector -fPIC ${PORTS_CFLAGS}
 LDFLAGS=-Wl,-z,now -Wl,-z,relro
 all: reference
 
-OBJS= crypto_scrypt-nosse.o sha256.o crypto_scrypt-hexconvert.o crypto-mcf.o modp_b64.o crypto-scrypt-saltgen.o crypto_scrypt-check.o crypto_scrypt-hash.o
+OBJS= crypto_scrypt-nosse.o sha256.o crypto_scrypt-hexconvert.o crypto-mcf.o b64.o crypto-scrypt-saltgen.o crypto_scrypt-check.o crypto_scrypt-hash.o
 
+endian.h: byteorder.c
+	$(CC) byteorder.c $(CFLAGS) -o byteorder
+	./byteorder > endian.h
 
-library: $(OBJS)
-	$(CC)  $(LDFLAGS) -shared -Wl,-soname,libscrypt.so.0 -Wl,--version-script=libscrypt.version -o libscrypt.so.0 -lc -lm  $(OBJS)
+library: endian.h $(OBJS)
+	$(CC)  $(LDFLAGS) -shared -Wl,--version-script=libscrypt.version -o libscrypt.so.${LIBscrypt_VERSION} -lc -lm  $(OBJS)
 	ar rcs libscrypt.a  $(OBJS)
 
 reference: library main.o
-	ln -s -f libscrypt.so.0 libscrypt.so
 	$(CC) -Wall -o reference main.o -Wl,-rpath=. -L.  -lscrypt
 
 clean:
-	rm -f *.o reference libscrypt.so* libscrypt.a
+	rm -f *.o reference libscrypt.so* libscrypt.a endian.h
 
 check: all
 	./reference
@@ -34,8 +36,7 @@ devtest:
 
 install: library
 	$(MAKE_DIR) $(DESTDIR) $(DESTDIR)$(PREFIX) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INCLUDEDIR)
-	$(INSTALL_DATA) -pm 0755 libscrypt.so.0 $(DESTDIR)$(LIBDIR)
-	cd $(DESTDIR)$(LIBDIR) && ln -s -f libscrypt.so.0 $(DESTDIR)$(LIBDIR)/libscrypt.so
+	$(INSTALL_DATA) -pm 0755 libscrypt.so.${LIBscrypt_VERSION} $(DESTDIR)$(LIBDIR)
 	$(INSTALL_DATA) -pm 0644 libscrypt.h  $(DESTDIR)$(INCLUDEDIR)
 
 install-static: libscrypt.a
